### PR TITLE
Introducing FSDP2 in documentation

### PR DESCRIPTION
**Pull request by nagpalar**
EKS:
- Introducing FSDP2 in documentation
- Added "What's New in FSDP2" section
- Docker build now has two targets: base (EKS) and hpto (HyperPod)
- Image tag bumped from pytorch2.5.1 to pytorch2.6.0
- Added HF_TOKEN to HyperPod CLI env vars
- Added FSDP2 implementation details, migration guide, and resources

Slurm:
- Introducing FSDP2 in documentation
- No other changes needed - train.py already uses FSDP2 APIs (fully_shard, MixedPrecisionPolicy, DTensor) under the hood

### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction. When we triage the issues, we will add labels to the issue like "Enhancement", "Bug" which should indicate to you that this issue can be worked on and we are looking forward to your PR. We would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/ai-on-sagemaker-hyperpod/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### More

- [ x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. 
- [ ] Yes, I have added a example to support my blueprint PR.
- [ x] Yes, I have updated the `website/docs` section for this feature.

### For Moderators

- [ x] E2E Test successfully complete before merge?
- [ ] I ran the proper security checks explained on the ML Frameworks wiki

### CODE REVIEW

File: website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md [hunk 1]
```diff
@@ -308,6 +334,65 @@ To stop the current training job, use the following command:
...
+model.to_empty(device=torch.device("cuda"))
+```
+
+**4. DTensor Parameters**
+All model parameters are now DTensor with Shard(0) placement, enabling:
+- Seamless optimizer integration
+- Communication-free sharded checkpoints
+- Easier parameter manipulation
+
+**5. Elastic Training Support**
```
Comment (mvinci12) [inline website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md]: thought no elastic training

File: website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md [hunk 2]
```diff
@@ -860,4 +946,27 @@ hyperpod list-jobs -n kubeflow
 {
  "jobs": []
 }
-```
\ No newline at end of file
+```
+
+
+## Migration from FSDP1 to FSDP2
```
Comment (mvinci12) [inline website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md]: lets this chunk remove and link huggingface docs on fsdp2 migration

**Review by mvinci12 — CHANGES_REQUESTED** [review]

File: website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md (hunk 2 above)
Comment (nagpalar) [reply website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md]: removed the same..

File: website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md (hunk 1 above)
Comment (nagpalar) [reply website/docs/00-eks-orchestration/training-and-fine-tuning/fsdp/fully-sharded-data-parallel.md]: Thank you for catching this again...... updated

**Review by mvinci12 — APPROVED** [review]
looks good
